vscode/oso/client/src/telemetry.ts

```diff
@@ -20,6 +20,28 @@ const distinct_id = hash(env.machineId);
 const MIXPANEL_PROJECT_TOKEN = 'd14a9580b894059dffd19437b7ddd7be';
 const mixpanel = Mixpanel.init(MIXPANEL_PROJECT_TOKEN, { protocol: 'https' });
 
+function telemetryEnabled() {
+  const setting = workspace
+    .getConfiguration('oso.polarLanguageServer.telemetry')
+    .get<'default' | 'on' | 'off' | undefined>('enabled');
+
+  // Check if user explicitly enabled or disabled telemetry.
+  if (setting === 'on') return true;
+  if (setting === 'off') return false;
+
+  // Otherwise, default to VSCode's telemetry setting.
+
+  // VSCode >=1.55
+  //
+  // https://code.visualstudio.com/updates/v1_55#_telemetry-enablement-api
+  if (env.isTelemetryEnabled !== undefined) return env.isTelemetryEnabled;
```

Comment:
I think you also need to look at the telemetry change event: https://code.visualstudio.com/api/references/vscode-api

See (`onDidChangeTelemetryEnabled`)

Reply:
This check happens every time before we record or send telemetry. That `onDidChangeTelemetryEnabled` event only matters if you're maintaining local state about whether or not telemetry is enabled & need to update that every time the user changes their telemetry settings. Since we check every time before we do anything, we get updates immediately.